### DISCUSSION TITLE
Expanded text contains internal edges

### DISCUSSION THREAD

**Discussion by abejfehr** [jscad/OpenJSCAD.org#1143] [Q&A]
I'm looking to expand some text and inset it on a shape. I've noticed that with a few characters, particularly numbers, it creates some overlapping sections, even if I union them.

This is some sample code:

```
  const labelSegments = vectorText({
    input: "123456",
    height: 3.5,
  });

  const labelPaths = labelSegments.map((s) => segmentToPath(s));

  const labelOutline = union(
    ...expand({ delta: 0.6, corners: "round" }, labelPaths)
  );
```

And here is what `labelOutline` looks like:

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/3917428/189576374-b3c45963-98ee-4a15-8228-9b0aa61445f8.png">

I don't see any other tools that could be used to create 1 outline per character, but I may have missed something. How can I avoid those internal lines that appear on the 1, 3, 5 and 6?

**Comment by z3dev** [comment]
This is the same problem as #782 , and cannot be solved easily.

however, there are other ways to assemble 2D outlines from the set of points (letters). Please see the example on text.

**Comment by z3dev** [comment]
@abejfehr There is also a little module that can be added to JSCAD projects, which can produce outlines from TTF fonts. You might be able to use this instead, after making some changes.

https://github.com/jscad-community/jscad-text

**Comment by hrgdavor** [comment]
@abejfehr keep an eye on V3, a new 2d boolean alg is being added (martinez) https://github.com/jscad/OpenJSCAD.org/pull/1177 and it will likely be much better at handling things like that.
